### DISCUSSION TITLE
Does rabbitmq client support MQTT

### DISCUSSION THREAD

**Discussion by fft164925585** [rabbitmq/rabbitmq-dotnet-client#1196] [General]
I want to use the RabbitMQ MQTT, but I am not sure if the rabbitmq client support this. Have some samples?

**Comment by michaelklishin** [comment]
If you want to use MQTT to connect to RabbitMQ, then use an MQTT client for .NET. Our team does not develop one and we do not implement multiple protocols in a single library.

**Comment by ogrim** [comment]
I just needed to send the same response to MQTT as well as AMQP, so turns out this works fine:

``` C#
_channel.BasicPublish("", replyQueuename, props, body); // amqp
_channel.BasicPublish("amq.topic", replyQueuename, props, body); // mqtt
```

According to https://www.rabbitmq.com/mqtt.html#implementation the MQTT plugin for RabbitMQ binds to `amq.topic` exchange by default and translates the queue name vs topic name under the hood. So `replyQueuename` in the code is `some.test` and for MQTT it ends up at `some/test` topic.

Now for consuming MQTT I just use [MQTTnet](https://www.nuget.org/packages/MQTTnet)

**Reply by michaelklishin** [reply]
It remains to be seen whether this would continue working after https://github.com/rabbitmq/rabbitmq-server/pull/5895 is merged. Likely it should because the routing part hasn't changed so far.

**Reply by ogrim** [reply]
Thanks for the info, I will make sure to pay attention to this :)
